tests/Aspire.Dashboard.Tests/Integration/DashboardClientAuthTests.cs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using Aspire.Dashboard.Configuration;
+using Aspire.Dashboard.Model;
+using Google.Protobuf.WellKnownTypes;
+using Aspire.ResourceService.Proto.V1;
+using Grpc.Core;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Xunit;
+using DashboardServiceBase = Aspire.ResourceService.Proto.V1.DashboardService.DashboardServiceBase;
+
+namespace Aspire.Dashboard.Tests.Integration;
+
+public sealed class DashboardClientAuthTests
+{
+    private const string ApiKeyHeaderName = "x-resource-service-api-key";
+
+    [Fact]
+    public async Task ConnectsToResourceService_Unsecured()
```

Comment:
Should there be HTTPS versions of these tests?

Reply:
On that topic, I thought there wasn't much point in testing client cert auth, since it was just adding a certificate to the handler, but certificate loading happens inside the client.

I think we either need to test the end-to-end with cert auth or split the cert loading out into a separate type and test loading cert from configuration independently.

Reply:
> Should there be HTTPS versions of these tests?

https://github.com/dotnet/aspire/pull/4621